src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ValueProviderResult.cs

```diff
@@ -188,22 +124,22 @@ public bool Equals(ValueProviderResult other)
         /// <inheritdoc />
         public override int GetHashCode()
         {
-            return ((string)this)?.GetHashCode() ?? 0;
+            return ToString().GetHashCode();
         }
 
         /// <inheritdoc />
         public override string ToString()
         {
-            return (string)this;
+            return Values.ToString();
         }
 
         /// <summary>
-        /// Gets an <see cref="Enumerator"/> for this <see cref="ValueProviderResult"/>.
+        /// Gets an <see cref="IEnumerator<string>"/> for this <see cref="ValueProviderResult"/>.
         /// </summary>
-        /// <returns>An <see cref="Enumerator"/>.</returns>
-        public Enumerator GetEnumerator()
+        /// <returns>An <see cref="IEnumerator<string>"/>.</returns>
+        public IEnumerator<string> GetEnumerator()
```

Comment:
If you're going to do this you can remove the explicit implementation for `IEnumerable<string>`